### DISCUSSION TITLE
[Flutter] 'Failed to install dependencies' on Codemagic, works fine locally

### DISCUSSION THREAD

**Discussion by pocotheo** [codemagic-ci-cd/codemagic-docs#1657] [Q&A]
Flutter build works fine locally on Visual Studio Code, but it returns 'failed to install dependencies' on Codemagic.

This is the error:
'Because <app> depends on flutter_localizations from sdk which depends on intl 0.18.0, intl 0.18.0 is required.
So, because <app> depends on intl 0.17.0, version solving failed.'
When I try to override the intl dependency to 0.18.0, the dependencies can be installed on Codemagic, but I get an error regarding some conflicts between some other dependencies.

I have my environment sdk set to  ">=2.12.0 <3.0.0" in my pubspec.yaml, but it seems it isn't taken into account when I try to deploy it on Codemagic on Flutter stable channel.

Can you please help me?

**Comment by dtrdic** [comment]
hey @pocotheo can you please share your build ID so I can take a look?

**Reply by pocotheo** [reply]
Hey @dtrdic, this is the build ID: 64634a8ec2e18f3ee5f39277.
Thank you for answering.

**Comment by pocotheo** [comment]
Hello @dtrdic ,

I can’t find your question on Github, so I’ll answer you here.
On my pubspec.yaml file I set the environment to: ">=2.12.0 <3.0.0”
On Codemagic, I use the stable channel.

Thanks,
Theodor

**Reply by patthipati** [reply]
Update environment to  sdk: '>=3.0.0 <4.0.0' in your pubspec.yaml 

OR

change the flutter version to 3.7.12 for you build to success

**Reply by pocotheo** [reply]
@patthipati but it builds on VSCode successfully (with ">=2.12.0 <3.0.0” set) . Why wouldn't it work on Codemagic?

**Reply by pocotheo** [reply]
I had to change the Codemagic environment to 3.3.10
